### PR TITLE
feat(msteams): save metric alert

### PR DESCRIPTION
**Pull request by scefali**
This PR adds the ability to save (but not send) a metric alert for the upcoming Microsoft Teams integration (which is hidden behind a feature flag). I also updated the check on the channel/user to be case insensitive.

Note, we are not handling some of the edge cases like when the username is not unique. We can add that in if/when that becomes necessary.

Here's what the UI looks like:

Screen Shot 2020-07-20 at 5 00 52 PM
Screen Shot 2020-07-20 at 5 01 06 PM

### CODE REVIEW

File: src/sentry/integrations/msteams/utils.py [hunk 1]
```diff
@@ -39,7 +39,9 @@ def get_channel_id(organization, integration_id, name):
         member_list = members.get("members")
         continuation_token = members.get("continuationToken")
 
-        filtered_members = list(filter(lambda x: x.get("name") == name, member_list))
+        filtered_members = list(
```
Comment (wedamija) [inline src/sentry/integrations/msteams/utils.py:42]: Lol, damn, does msteams not provide a proper search api either?

**Review by wedamija — APPROVED** [review]
Just confirming: This won't appear in the UI unless the user has the integration enabled, and they can't yet because they won't have the feature flag

File: src/sentry/integrations/msteams/utils.py (hunk 1 above)
Comment (scefali) [reply src/sentry/integrations/msteams/utils.py:42]: @wedamija No, that would be too easy and smart

**Comment by scefali** [comment]
> Just confirming: This won't appear in the UI unless the user has the integration enabled, and they can't yet because they won't have the feature flag

It shouldn't the same way Slack wouldn't appear unless you have Slack installed.

**Pull request merged**